### ISSUE TITLE
[dualtor] Standby ToR re-encapsulate the encap traffic back to T1s

### ISSUE THREAD

**Issue by lolyu** [Azure/sonic-buildimage#6841]
#### Description
For a ToR interface that is in standby mode, if we send some encapsulated packets from the other ToR to this ToR, it should not decapsulate, re-encapsulate those packets back to T1s.

#### Steps to reproduce the issue:
1. Config the upper ToR's `Ethernet8`  as standby via swssconfig
```
admin@str2-7050cx3-acs-08:/etc/sonic$ show muxcable status Ethernet8
PORT       STATUS    HEALTH
---------  --------  --------
Ethernet8  standby   HEALTHY
admin@str2-7050cx3-acs-08:/etc/sonic$ show muxcable config Ethernet8
SWITCH_NAME          PEER_TOR
-------------------  ----------
str2-7050cx3-acs-09  10.1.0.33
port       state    ipv4            ipv6
---------  -------  --------------  ----------------
Ethernet8  auto     192.168.0.4/32  fc02:1000::4/128
```
2. Send an encapsulated packet from the lower ToR to the upper ToR with the inner packet destinated to server `192.168.0.4`
```
(Pdb) encapsulated_packet.show()
###[ Ethernet ]###
  dst       = d4:af:f7:4d:a5:4c
  src       = 1a:a5:06:36:f5:00
  type      = 0x800
###[ IP ]###
     version   = 4
     ihl       = None
     tos       = 0x70
     len       = None
     id        = 1
     flags     =
     frag      = 0
     ttl       = 255
     proto     = ipencap
     chksum    = None
     src       = 10.1.0.33
     dst       = 10.1.0.32
     \options   \
###[ IP ]###
        version   = 4
        ihl       = None
        tos       = 0x70
        len       = None
        id        = 1
        flags     =
        frag      = 0
        ttl       = 4
        proto     = hopopt
        chksum    = None
        src       = 1.1.1.1
        dst       = 192.168.0.4
        \options   \
###[ Raw ]###
           load      = '\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !"#$%&\'()*+,-./0123456789:;<=>?@A'
```

#### Describe the results you received:
The packet could be confirmed that it has been decapsulated and re-encapsulated back to T1s via one of those ptf T1 intercepted ports:
```
root@c47680ead3ca:~# tcpdump src 10.1.0.32 and dst 10.1.0.33 -i eth69 -nnevv
tcpdump: listening on eth69, link-type EN10MB (Ethernet), capture size 262144 bytes
08:26:35.226541 d4:af:f7:4d:a5:4c > 52:54:00:4c:3c:07, ethertype IPv4 (0x0800), length 120: (tos 0x0, ttl 3, id 6009, offset 0, flags [none], proto IPIP (4), length 106)
    10.1.0.32 > 10.1.0.33: (tos 0x70, ttl 3, id 1, offset 0, flags [none], proto Options (0), length 86)
    1.1.1.1 > 192.168.0.4:  ip-proto-0 66
```
And the packet could be confirmed received by the corresponding ptf port for the server `192.168.0.4`
```
root@c47680ead3ca:~# tcpdump src 1.1.1.1 -i eth4 -nnev
tcpdump: listening on eth4, link-type EN10MB (Ethernet), capture size 262144 bytes
08:26:35.234558 00:aa:bb:cc:dd:ee > 92:39:31:c2:a9:04, ethertype IPv4 (0x0800), length 100: (tos 0x70, ttl 2, id 1, offset 0, flags [none], proto Options (0), length 86)
    1.1.1.1 > 192.168.0.4:  ip-proto-0 66
```

#### Describe the results you expected:
The packet should not be re-encapsulated

```
admin@str2-7050cx3-acs-08:/etc/sonic$ show version

SONiC Software Version: SONiC.20201231.beta.6
Distribution: Debian 10.8
Kernel: 4.19.0-9-2-amd64
Build commit: 35136c4a2
Build date: Wed Feb 10 10:20:44 UTC 2021
Built by: sonicbld@w1

Platform: x86_64-arista_7050cx3_32s
HwSKU: Arista-7050CX3-32S-D48C8
ASIC: broadcom
ASIC Count: 1
Serial Number: JPW20280924
Uptime: 08:32:34 up  2:24,  1 user,  load average: 2.87, 3.47, 3.70

Docker images:
REPOSITORY                 TAG                 IMAGE ID            SIZE
docker-syncd-brcm          20201231.beta.6     9c3f51ac010f        680MB
docker-syncd-brcm          latest              9c3f51ac010f        680MB
docker-teamd               20201231.beta.6     78abee0efc28        408MB
docker-teamd               latest              78abee0efc28        408MB
docker-router-advertiser   20201231.beta.6     7eb52303b062        398MB
docker-router-advertiser   latest              7eb52303b062        398MB
docker-platform-monitor    20201231.beta.6     c4f803d26565        607MB
docker-platform-monitor    latest              c4f803d26565        607MB
docker-snmp                20201231.beta.6     f2dd4f37373a        438MB
docker-snmp                latest              f2dd4f37373a        438MB
docker-database            20201231.beta.6     348a702ce545        398MB
docker-database            latest              348a702ce545        398MB
docker-lldp                20201231.beta.6     5f25dd0ebec2        438MB
docker-lldp                latest              5f25dd0ebec2        438MB
docker-orchagent           20201231.beta.6     48fe21b602a5        426MB
docker-orchagent           latest              48fe21b602a5        426MB
docker-dhcp-relay          20201231.beta.6     3f5724c79997        405MB
docker-dhcp-relay          latest              3f5724c79997        405MB
docker-sonic-telemetry     20201231.beta.6     17efc7572c1d        472MB
docker-sonic-telemetry     latest              17efc7572c1d        472MB
docker-mux                 20201231.beta.6     7765777aeebc        452MB
docker-mux                 latest              7765777aeebc        452MB
docker-fpm-frr             20201231.beta.6     e9f00d5adb3d        425MB
docker-fpm-frr             latest              e9f00d5adb3d        425MB
docker-acms                20201231.beta.6     34a30c7796ae        197MB
docker-acms                latest              34a30c7796ae        197MB
k8s.gcr.io/pause           3.2                 80d28bedfe5d        683kB
```

**Comment by lolyu** [comment]
@prsunny, could you please help confirm this?

**Comment by prsunny** [comment]
This should be fixed in the latest SAI drop

**Comment by lolyu** [comment]
Still hit this issue with the following image
```
admin@str2-7050cx3-acs-08:~$ show version | head -n5

SONiC Software Version: SONiC.20201231.beta.11
Distribution: Debian 10.8
Kernel: 4.19.0-12-2-amd64
Build commit: f37ce1d2b
admin@str2-7050cx3-acs-08:~$ docker exec -it syncd bash
root@str2-7050cx3-acs-08:/# dpkg -s libsaibcm | head -n10
Package: libsaibcm
Status: install ok installed
Priority: extra
Section: libs
Installed-Size: 233425
Maintainer: Guohan Lu <gulv@microsoft.com>
Architecture: amd64
Source: saibcm
Version: 4.3.3.1
Provides: libsai
```

**Comment by lolyu** [comment]
The 20000 encapsulated packets got passed to the active ToR:
```
admin@str2-7050cx3-acs-10:~$ portstat
Last cached time was 2021-03-10 05:28:31.250663
      IFACE    STATE    RX_OK      RX_BPS    RX_UTIL    RX_ERR    RX_DRP    RX_OVR    TX_OK      TX_BPS    TX_UTIL    TX_ERR    TX_DRP    TX_OVR
-----------  -------  -------  ----------  ---------  --------  --------  --------  -------  ----------  ---------  --------  --------  --------
  Ethernet0        U        0    0.00 B/s      0.00%         0         0         0    1,444  706.98 B/s      0.00%         0         0         0
  Ethernet2        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
  Ethernet4        U        0    0.00 B/s      0.00%         0         0         0    1,444  706.98 B/s      0.00%         0         0         0
  Ethernet6        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
  Ethernet8        U        0    0.00 B/s      0.00%         0         0         0    1,444  706.98 B/s      0.00%         0         0         0
 Ethernet10        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet12        U        0    0.00 B/s      0.00%         0         0         0    1,444  706.98 B/s      0.00%         0         0         0
 Ethernet14        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet16        U        0    0.00 B/s      0.00%         0         0         0    1,444  706.98 B/s      0.00%         0         0         0
 Ethernet18        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet20        U        0    0.00 B/s      0.00%         0         0         0    1,444  706.98 B/s      0.00%         0         0         0
 Ethernet22        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet24        U   20,018  17.20 KB/s      0.00%         0         0         0       10   13.79 B/s      0.00%         0         0         0
 Ethernet28        U       10   10.57 B/s      0.00%         0         0         0       18   18.75 B/s      0.00%         0         0         0
 Ethernet32        U       10   10.57 B/s      0.00%         0         0         0   20,010  17.20 KB/s      0.00%         0         0         0
 Ethernet36        U       42   31.24 B/s      0.00%         0         0         0       42   34.49 B/s      0.00%         0         0         0
 Ethernet40        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.53 B/s      0.00%         0         0         0
 Ethernet42        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet44        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.53 B/s      0.00%         0         0         0
 Ethernet46        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet48        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.53 B/s      0.00%         0         0         0
 Ethernet50        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet52        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.53 B/s      0.00%         0         0         0
 Ethernet54        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet56        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
 Ethernet58        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet60        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
 Ethernet62        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet64        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
 Ethernet66        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet68        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
 Ethernet70        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet72        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
 Ethernet74        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet76        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
 Ethernet78        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet80        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
 Ethernet82        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet84        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
 Ethernet86        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
 Ethernet88        U       18   15.53 B/s      0.00%         0         0         0       10   13.82 B/s      0.00%         0         0         0
 Ethernet92        U       10   10.57 B/s      0.00%         0         0         0       18   18.78 B/s      0.00%         0         0         0
 Ethernet96        U       18   15.53 B/s      0.00%         0         0         0       10   13.82 B/s      0.00%         0         0         0
Ethernet100        U       10   10.57 B/s      0.00%         0         0         0       18   18.78 B/s      0.00%         0         0         0
Ethernet104        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
Ethernet106        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
Ethernet108        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
Ethernet110        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
Ethernet112        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
Ethernet114        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
Ethernet116        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
Ethernet118        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
Ethernet120        U        0    0.00 B/s      0.00%         0         0         0    1,443  706.56 B/s      0.00%         0         0         0
Ethernet122        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
Ethernet124        U        0    0.00 B/s      0.00%         0         0         0    1,453  711.41 B/s      0.00%         0         0         0
Ethernet126        X        0    0.00 B/s      0.00%         0         0         0        0    0.00 B/s      0.00%         0         0         0
```

**Comment by lguohan** [comment]
so, the encapped packet is not dropped on the standby tor, and still got send back to the active tor?

**Comment by lolyu** [comment]
@lguohan, yes, those packets are not dropped.

**Comment by prsunny** [comment]
@lolyu, this shall be fixed in the next image. There will still be an outstanding issue with v6inv4 which is being tracked via CSP

**Comment by prsunny** [comment]
This is resolved in SAI 4.3.3.4. Verified for both IPv4 and IPv6 traffic.

**Comment by theasianpianist** [comment]
Still seen as per #7602 . Continuing to track on this thread

**Comment by prsunny** [comment]
Opened CSP Case CS00012192505 - [4.3] Re-encap IPinIP decap packets

**Comment by prsunny** [comment]
Closing this as fix is available as part of 4.3.3.9
